frontend/src/pages/ShopList.jsx

```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+
+export default function ShopList() {
+  const [results, setResults] = useState([]);
+  const [types, setTypes] = useState([]);
+
+  const getResults = () => {
+    axios
+      .get(
+        `${import.meta.env.VITE_BACKEND_URL ?? "http://localhost:5000"}/shops`
+      )
+      .then((response) => response.data)
+      .then((data) => {
+        setResults(data);
+      });
+  };
```

Comment:
this call should get directly the shop with the types, using a join. then the data should be restructured with something like that :

    setResults(
        data.reduce((shopsByTypes, shop) => {
            if (shopByTypes[shop.type] == null) {
                shopByTypes[shop.type] = [];
            }
            shopByTypes[shop.type].push(shop);
        }, {})
    );